university-of-pretoria-harvard-theology-religion.csl

```diff
@@ -0,0 +1,341 @@
+          </if>
+          <else-if type="webpage" match="any">
+            <text variable="container-title" form="short"/>
+          </else-if>
+          <else>
+            <text variable="title" form="short" quotes="true" text-case="capitalize-first"/>
+          </else>
+        </choose>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-count">
+    <names variable="author">
+      <name form="count"/>
+    </names>
+  </macro>
+  <macro name="date-issued">
+    <choose>
+      <if type="article-journal" match="none">
+        <text macro="date-issued-year"/>
+      </if>
+      <else-if variable="page" match="none">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else-if is-numeric="page">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else>
+        <text term="in press" prefix="(" suffix=")"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="date-issued-no-parentheses">
+    <choose>
+      <if type="article-journal" match="none">
+        <text macro="date-issued-year"/>
+      </if>
+      <else-if variable="page" match="none">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else-if is-numeric="page">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else>
+        <text term="in press"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="date-issued-year">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="access">
+    <choose>
+      <if variable="DOI">
+        <text variable="DOI" prefix="doi.org/"/>
+      </if>
+      <else-if variable="URL">
+        <group delimiter=", ">
+          <text macro="date-access"/>
+          <group>
+            <text term="available at" suffix=" "/>
+            <text variable="URL"/>
+          </group>
+        </group>
+      </else-if>
+      <else-if variable="source accessed" match="all">
+        <group delimiter=", ">
+          <text macro="date-access"/>
+          <text variable="source"/>
+        </group>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="date-access">
+    <date variable="accessed" prefix="viewed ">
+      <date-part name="day" suffix=" "/>
+      <date-part name="month"/>
+      <date-part name="year" prefix=" "/>
+    </date>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="bill book graphic legal_case legislation motion_picture report song" match="any">
+        <text variable="title" font-style="italic"/>
+      </if>
+      <else>
+        <text variable="title"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="book-details">
+    <group delimiter=", ">
+      <group delimiter=" ">
+        <number variable="edition" form="ordinal"/>
```

Comment:
worthwhile having coding for non-numeric editions in the style -- that's fairly common in user data (most existing style have an `if is-numeric="edition"` test)